pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -184,42 +197,42 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	// from odo standpoint, these components contain yaml manifest of an odo service or an odo link
 	k8sComponents, err := devfile.GetKubernetesComponentsToPush(a.Devfile)
 	if err != nil {
-		return errors.Wrap(err, "error while trying to fetch service(s) from devfile")
+		return fmt.Errorf("error while trying to fetch service(s) from devfile: %w", err)
 	}
 
 	// validate if the GVRs represented by Kubernetes inlined components are supported by the underlying cluster
 	err = service.ValidateResourcesExist(a.Client, k8sComponents, a.Context)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to validate if the GVR is supported by the cluster: %w", err)
 	}
 
 	// create the Kubernetes objects from the manifest and delete the ones not in the devfile
-	err = service.PushKubernetesResources(a.Client, k8sComponents, labels, a.Context)
+	err = service.PushKubernetesResources(a.Client, k8sComponents, labels, annotations, a.Context)
 	if err != nil {
-		return errors.Wrap(err, "failed to create service(s) associated with the component")
+		return fmt.Errorf("failed to create service(s) associated with the component: %w", err)
 	}
 
 	isMainStorageEphemeral := a.prefClient.GetEphemeralSourceVolume()
 	err = a.createOrUpdateComponent(componentExists, parameters.EnvSpecificInfo, isMainStorageEphemeral)
 	if err != nil {
-		return errors.Wrap(err, "unable to create or update component")
+		return fmt.Errorf("unable to create or update component: %w", err)
 	}
 
 	a.deployment, err = a.Client.WaitForDeploymentRollout(a.deployment.Name)
 	if err != nil {
-		return errors.Wrap(err, "error while waiting for deployment rollout")
+		return fmt.Errorf("error while waiting for deployment rollout: %w", err)
 	}
 
 	// Wait for Pod to be in running state otherwise we can't sync data or exec commands to it.
 	pod, err := a.getPod(true)
 	if err != nil {
-		return errors.Wrapf(err, "unable to get pod for component %s", a.ComponentName)
+		return fmt.Errorf("unable to get pod for component %s: %w", a.ComponentName, err)
 	}
 
 	// list the latest state of the PVCs
 	pvcs, err := a.Client.ListPVCs(fmt.Sprintf("%v=%v", "component", a.ComponentName))
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to list the state of all PVCs: %w", err)
```

Comment:
Thank you for this! (and all the others)

Reply:
What is purpuse of this? 
Why are we  removing `error.Wrapf` from this file, but using it in other files?

Reply:
More info on https://github.com/redhat-developer/odo/pull/5468#discussion_r822764627